compiler/vm/vmdef.nim

```diff
@@ -121,17 +319,61 @@ type
     slots*: ptr UncheckedArray[TFullReg]
     currentException*: PNode
     currentLineInfo*: TLineInfo
+
+    # XXX: These are only here as a temporary measure until I continue 
+    # to work on callback handling
+    typeCache*: ptr TypeInfoCache
+    mem*: ptr VmMemoryManager
+
   VmCallback* = proc (args: VmArgs) {.closure.}
 
+  SlotStorage* = object ## A fixed size container for storing registers 
+                        ## that also allows for shallow copies
+    slots: ptr UncheckedArray[TFullReg]
+    length: int
+    owner: bool ## Indicates whether an instance of `SlotStorage` is the owner of the backing memory
+
+  VmMemoryManager* = object
+    # XXX: the current implementation of the memory manager is overly simple and only temporary
+    regions*: seq[tuple[typ: PVmType, count: int, start: int, len: int]]
+    freed*: seq[tuple[start: int, len: int]]
+
+  TypeInfoCache* = object ## An append-only cache for storing all `VmType`s
+                          ## and `PType` to `PVmType` mappings
+    lut*: Table[ItemId, PVmType]
+
+    types*: seq[PVmType]
```

Comment:
Please document these two fields

Reply:
Will do. Documentation is for the most part still missing. I'm planning to do that once everything is in place, since a lot of smaller things might still change and I don't want to document things that are getting thrown away in the end (inside the bounds of this PR)